### PR TITLE
2nd Assignment by 202503758 (Choi BIhwan)

### PR DESCRIPTION
**Pull request by bihwanchoi**
`submissions\assignment2\202503758\problem1.py` 제출합니다.

### CODE REVIEW

**Comment by hist0613** [comment]
아직 과제가 정식으로 나오지 않았으니 나오면 리뷰하도록 하겠습니다 ㅎㅎ

**Comment by bihwanchoi** [comment]
교수님 과제는 다 진행했는데 혹시 `grader.py`는 어떤 파일인지 알 수 있을까요?

**Comment by hist0613** [comment]
- `grader.py`는 과제 기간 중에는 비공개입니다. 과제가 끝난 후에 공유가 될 예정입니다 :)
- `grader.py`로 확인해 본 결과 전반적으로 다 작성을 잘 해주신 것 같은데, problem 4 에서 `train_test_split` 에 예외처리가 하나 잘못된 부분이 있습니다 :)

File: submissions/202503758/assignment2/problem1/README.md
Comment (hist0613) [inline submissions/202503758/assignment2/problem1/README.md:1]: 요거 잘뭇 변경된 것 같네요?

File: submissions/202503758/assignment2/problem3/main.py [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+"""
+
+from collections import Counter
+
+def count_tokens(tokens: list[str]) -> dict[str, int]:
+    # 힌트:
+    # 1) 빈 딕셔너리 생성: d = {}
+    # 2) 각 토큰을 순회하면서 카운트: d[token] = d.get(token, 0) + 1
+    # 3) 또는 collections.Counter 사용 가능 (하지만 직접 구현도 간단함)
+    return Counter(tokens)
```
Comment (hist0613) [inline submissions/202503758/assignment2/problem3/main.py:15]: 이 부분은 `Counter` 사용하지 말고 직접 구현해주세요!

File: submissions/202503758/assignment2/problem4/dsops/stats/labels.py
Comment (hist0613) [inline submissions/202503758/assignment2/problem4/dsops/stats/labels.py:1]: 문제 spec에서 명시가 된 경우를 제외하고, 라이브러리를 사용하지 말고 직접 구현해주세요~

File: submissions/202503758/assignment2/problem3/main.py (hunk 1 above)
Comment (hist0613) [reply submissions/202503758/assignment2/problem3/main.py:15]: 아 제가 과제에 명시를 해놨네요ㅠ 사용하셔도 괜찮습니다!

File: submissions/202503758/assignment2/problem2/textops/clean/filters.py [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    # 힌트:
+    # 1) s.lower() - 소문자 변환
+    # 2) re.sub(r"\s+", " ", s) - 모든 연속 공백을 단일 공백으로
+    # 3) s.strip() - 앞뒤 공백 제거
+    # 4) string.punctuation에서 특정 문자 제외하고 제거
+    # 5) set 연산을 활용해서 keep = {"'", "-"}, 나머지는 제거
+    s = s.lower()
+    s = re.sub(r"\s+", " ", s)
+    s = s.strip()
+    remove = set(string.punctuation) - {"'", "-"}
```
Comment (hist0613) [inline submissions/202503758/assignment2/problem2/textops/clean/filters.py:21]: good :)

**Review by hist0613 — APPROVED** [review]
LGTM :) 본문 내용만 채워주세요
